src/index/base.cpp

```diff
@@ -67,7 +67,7 @@ bool BaseIndex::Init()
         SetBestBlockIndex(m_chainstate->m_blockman.FindForkInGlobalIndex(active_chain, locator));
     }
     m_synced = m_best_block_index.load() == active_chain.Tip();
-    if (!m_synced) {
+    if (!m_synced && (fPruneMode || fHavePruned) && !fReindex) {
```

Comment:
I did not do performance tests on this change, it will probably not be huge but if we can get it, why not? But my main motivation is that I think that these checks are not necessary unless the conditions hold. I think it makes code easier to understand if sections that are not relevant are skipped explicitly. But I can see how this might be missing a comment for clarity, happy to add that if you agree.

Reply:
I agree with you that the line is harder to understand after the change but in exchange for that, it makes explicit that the code inside the block is only relevant and needs to run if this evaluates to true. Let's wait for more feedback as a tie-breaker, of course, I am happy to remove it if others agree with you :)

Reply:
> it makes explicit that the code inside the block is only relevant and needs to run if this evaluates to true

I don't have a strong opinion but i agree with this statement and think the change should stay.

Reply:
ISTM there should be a demonstrated clear motivation for the commit and documentation of that in both places.